File: flutter_client_1771020152/lib/models/menu_item.dart
class MenuItem {
  final int id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final bool isVegetarian;
  final bool isSpicy;

  MenuItem({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.isVegetarian,
    required this.isSpicy,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['Id'] ?? 0, // Chú ý: Web API .NET thường trả về chữ hoa đầu (Id, Name)
      name: json['Name'] ?? 'No Name',
      description: json['Description'] ?? '',
      price: (json['Price'] ?? 0).toDouble(),
      imageUrl: json['ImageUrl'] ?? '',
      isVegetarian: json['IsVegetarian'] ?? false,
      isSpicy: json['IsSpicy'] ?? false,
    );
  }
}

File: web_api_1771020152/web_api_1771020152/Models/MenuItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace web_api_1771020152.Models // Đảm bảo đúng namespace của bạn
{
    public class MenuItem
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string Description { get; set; } // Thêm dòng này

        public string Category { get; set; }

        public decimal Price { get; set; }

        public string ImageUrl { get; set; }

        public int PreparationTime { get; set; } // Thêm dòng này (Thời gian chế biến)

        public bool IsVegetarian { get; set; } = false;
        public bool IsSpicy { get; set; } = false;
        public bool IsAvailable { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}